volatility/framework/plugins/windows/netlist.py

```diff
@@ -0,0 +1,544 @@
+                                         "Neither UdpPortPool nor UdpCompartmentSet found in {} table".format(tcpip_symbol_table))
+
+        vollog.debug("Found PortPools @ 0x{:x} (TCP) && 0x{:x} (UDP)".format(upp_addr, tpp_addr))
+        return upp_addr, tpp_addr
+
+    @classmethod
+    def list_sockets(cls,
+                       context: interfaces.context.ContextInterface,
+                       layer_name: str,
+                       nt_symbols: str,
+                       net_symbol_table: str,
+                       tcpip_module: interfaces.objects.ObjectInterface,
+                       tcpip_symbol_table: str) -> \
+            Iterable[interfaces.objects.ObjectInterface]:
+        """Lists all UDP Endpoints, TCP Listeners and TCP Endpoints in the primary layer that
+        are in tcpip.sys's UdpPortPool, TcpPortPool and TCP Endpoint partition table, respectively.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            nt_symbols: The name of the table containing the kernel symbols
+            net_symbol_table: The name of the table containing the tcpip types
+            tcpip_module: The created vol Windows module object of the given memory image
+            tcpip_symbol_table: The name of the table containing the tcpip driver symbols
+
+        Returns:
+            The list of network objects from the `layer_name` layer's `PartitionTable` and `PortPools`
+        """
+
+        tcpip_module_offset = tcpip_module.DllBase
+
+        pointer_length = context.symbol_space.get_type(net_symbol_table + constants.BANG + "pointer").size
+
+        # first, TCP endpoints by parsing the partition table
+        for endpoint in cls.parse_partitions(context, layer_name, net_symbol_table, tcpip_symbol_table, tcpip_module_offset, pointer_length):
+            yield endpoint
+
+        # then, towards the UDP and TCP port pools
+        # first, find their addresses
+        upp_addr, tpp_addr = cls.find_port_pools(context, layer_name, net_symbol_table, tcpip_symbol_table, tcpip_module_offset, pointer_length)
+
+        # create port pool objects at the detected address and parse the port bitmap
+        upp_obj = context.object(net_symbol_table + constants.BANG + "_INET_PORT_POOL", layer_name = layer_name, offset = upp_addr)
+        udpa_ports = cls.parse_bitmap(context, layer_name, upp_obj.PortBitMap.Buffer, upp_obj.PortBitMap.SizeOfBitMap // 8)
+
+        tpp_obj = context.object(net_symbol_table + constants.BANG + "_INET_PORT_POOL", layer_name = layer_name, offset = tpp_addr)
+        tcpl_ports = cls.parse_bitmap(context, layer_name, tpp_obj.PortBitMap.Buffer, tpp_obj.PortBitMap.SizeOfBitMap // 8)
+
+        # given the list of TCP / UDP ports, calculate the address of their respective objects and yield them.
+        for port in tcpl_ports:
+            if port == 0:
```

Comment:
I actually think I'd prefer the verbose way, because I found it often that the bit at index 0 (which would correlate with port 0) is set to 1. I dislike about the pythonic way that it mushes together 0 and None, because in this scenario a value of 0 is actually expected and I'd like to explicitly skip that. We do not expect a value of None, so no special handling for that.

However, I've changed it and added comments to clarify - I think this is better in the long run ;)

Reply:
So I don't want to police stylistic choices where it doesn't really have an impact!  5:S  Happy for you to end up doing it whichever way you think is best.  5:)